File: src/commands/admin-commands/stop.ts
import { Discord, Slash } from "discordx";
import type { CommandInteraction } from "discord.js";
import { requireAdmin } from "../../helper/adminGuard.js";

@Discord()
export class Stop {
  @Slash({ name: "stop", description: "Stop the bot process" })
  async stop(interaction: CommandInteraction): Promise<void> {
    if (!requireAdmin(interaction)) return;

    await interaction.reply({ content: "Shutting down...", ephemeral: true });
    process.exit(0);
  }
}

File: unused/admin/stop.py
import sys
import discord
from discord.ext import commands
from utils.admin_guard import admin_check


class Stop(commands.Cog):
    @discord.app_commands.command(name="stop", description="Stop the bot process")
    @admin_check()
    async def stop(self, interaction: discord.Interaction):
        await interaction.response.send_message("Shutting down...", ephemeral=True)
        sys.exit(0)


async def setup(bot):
    await bot.add_cog(Stop())
